packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -1343,9 +1386,11 @@ class _DragInfo extends Drag {
     index = item.index;
     child = item.widget.child;
     capturedThemes = item.widget.capturedThemes;
-    dragPosition = initialPosition;
     dragOffset = itemRenderBox.globalToLocal(initialPosition);
     itemSize = item.context.size!;
+    _rawDragPosition = initialPosition;
+    boundary = listState.widget.dragBoundaryProvider?.call(listState.context);
```

Comment:
Have you considered having `ReorderableList` inherit the drag boundary through `forRectMaybeOf` by default instead of passing it down as a parameter? Or maybe it should do both, inherit by default, and have a parameter that overrides the default inherited provider.

Reply:
I'm also curious if there is any native behavior we have to compare to on other platforms (if a similar component exists).

Reply:
Modified to default to using `DragBoundary.forRectMaybeOf`.